contrib/src/instructions/bitwise.rs

```diff
@@ -0,0 +1,53 @@
+use symbolic_stack_machines_core::{
+    constraint::Constraint,
+    instructions::{AbstractExecRecord, AbstractInstruction, EnvExtension, InstructionResult},
+    memory::Mem,
+    stack::{Stack, StackOpRecord, StackRecord},
+};
+
+// The MachineEq trait is necessary because the type of equality used in the machine 
+// may be different than the type of equality used outside of the machine, and
+// `PartialEq` types the result of == as boolean.
+// 
+// Using `MachineEq` lets us use our own separate predicate type which in the case
+// where the machine holds symbolic values can in example be a symbolic boolean.
+pub trait MachineEq {
+    type Pred;
+
+    fn machine_eq(&self, other: &Self) -> Self::Pred;
+    fn machine_ite(pred: Self::Pred, then: Self, xelse: Self) -> Self;
+}
```

Comment:
This trait will probably be placed in a different module (maybe somewhere in core?) but `ISZERO` is the only uses for it rn.